manager/repository/repository_test.go

```diff
@@ -41,15 +42,15 @@ func TestRepositoryGetFailsWithNonExistentDeployment(t *testing.T) {
 	}
 }
 
-func TestRepositoryCreateDeploymentWorks(t *testing.T) {
+func createDeploymentWithManifests(t *testing.T, count int) {
```

Comment:
Seems like this should have a test in the name somewhere, since it doesn't only createDeployment, but also validates and tests lists, etc.?